components/PostCard.tsx

```diff
@@ -0,0 +1,67 @@
+import {  Box, Image, Text, Flex } from "@chakra-ui/react";
+import { useBookQuery } from "@lib/hooks/book";
+import { PostWithUser } from "@lib/services/post";
+import { useRouter } from "next/router";
+import UserWithAvatar from "./UserWithAvatar";
+
+export default function PostCard(post: PostWithUser) {
+  const { id, price, description, user, bookId, imageUrl } = post;
+  const { data: book } = useBookQuery(bookId);
+  const router = useRouter();
+
+  const handleClick = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    router.push("/post/" + id)
+  }
+
+  return (
+    <Flex
+      w="350px"
+      background='white'
+      overflow='hidden'
+      borderRadius='lg' 
+      mb='3'
+      mr='3'
+      direction='row'
+      shadow='md'
+      _hover={{ shadow: 'xl' }}
+      transition='0.3s'
+      cursor='pointer'
+      onClick={handleClick}
+    >
+      <Image 
+        maxH='200px'
+        objectFit='contain'
+        src={imageUrl || "https://via.placeholder.com/150"} 
+        fallbackSrc="https://via.placeholder.com/150"
```

Comment:
Can we put `book.imageUrl` (and maybe also `book.googleBooks[...]`) as the fallback image? Also checkout the NextJS image component.